telegram/ext/conversationhandler.py

```diff
@@ -423,6 +444,37 @@ def _get_key(self, update: Update) -> Tuple[int, ...]:
 
         return tuple(key)
 
+    def _resolve_promise(self, state: Tuple) -> object:
+        old_state, new_state = state
+        try:
+            res = new_state.result(0)
+            res = res if res is not None else old_state
+        except Exception as exc:
+            self.logger.exception("Promise function raised exception")
+            self.logger.exception("%s", exc)
+            res = old_state
+        finally:
+            if res is None and old_state is None:
+                res = self.END
+        return res
+
+    def _schedule_job(
+        self,
+        new_state: object,
+        dispatcher: 'Dispatcher',
+        update: Update,
+        context: Optional[CallbackContext],
+        conversation_key: Tuple[int, ...],
+    ) -> None:
+        if new_state != self.END:
+            self.timeout_jobs[
+                conversation_key
+            ] = dispatcher.job_queue.run_once(  # type:ignore[union-attr]
```

Comment:
just curiosity: Why are those type ignore comments necessary?

Reply:
Mypy was complaining about `Optional[JobQueue]` not having `run_once` attr and invalid type of `self.conversation_timeout`..
![IMG_20210321_171842_329](https://user-images.githubusercontent.com/58552395/111903695-8905f400-8a69-11eb-8141-f5492aa6361e.jpg)

Reply:
Ah, I see - The checks for `conversation_timeout` and `dispatcher.job_queue` happen outside of the function. Still, I think we can improve a bit:

* `if disptacher.job_queue is None` in `handle_update` (line 599), make a log on warning level that timeouts can't work without jobqueue
* Correct the annotation for the `conversation_timeout` argument of `__init__` to `Union[float, timedelta]` and for the property to `Optional[Union[float, timedelta]]`
* Add a comment that both `dispatcher.job_queue` and `conversation_timeout` are checked before calling `_schedule_timeout`, so future-us don't worry about it :)